### PR TITLE
[vSphere] Add new resourcepool datatastream

### PR DESCRIPTION
**Pull request by niraj-elastic**
- Enhancement

## What does this PR do?

- Added 1 data stream (Resourcepool Metrics).
- Added data collection metrics for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added dashboards and visualizations.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/vsphere) directory.
- Run the following command to run tests. elastic-package test

## TSDS tool kit output
unnamed (9)

## Related issues

- Closes https://github.com/elastic/obs-integration-team/issues/42

### CODE REVIEW

File: packages/vsphere/data_stream/resourcepool/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+title: "vSphere Resourcepool metrics"
+type: metrics
+streams:
+  - input: vsphere/metrics
+    title: vSphere resourcepool metrics
+    description: vSphere resourcepool metrics
+elasticsearch:
+  index_mode: "time_series"
```
Comment (lucian-ioan) [inline packages/vsphere/data_stream/resourcepool/manifest.yml:8]: Please provide a screenshot with results from [TSDB-migration-test-kit](https://github.com/elastic/TSDB-migration-test-kit) before enabling (after adding metric_type to the fields).

File: packages/vsphere/manifest.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 title: VMware vSphere
 format_version: "3.0.2"
 name: vsphere
-version: "1.13.0"
+version: "1.14.0"
```
Comment (lucian-ioan) [inline packages/vsphere/manifest.yml:4]: Use version: "1.14.0-next" as per this [issue](https://github.com/elastic/obs-integration-team/issues/22).

File: packages/vsphere/data_stream/resourcepool/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      type: group
+      fields:
+        - name: count
+          type: long
+          description: >
+            Number of virtual machines on the resource pool.
+        - name: names
+          type: keyword
+          description: >
+            Names of virtual machines on the resource pool.
```
Comment (lucian-ioan) [inline packages/vsphere/data_stream/resourcepool/fields/fields.yml:115]: Add metric_type to all these fields.

File: packages/vsphere/changelog.yml [hunk 4]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.14.0"
+  changes:
+    - description: Add resourcepool datastream.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/10996
```
Comment (lucian-ioan) [inline packages/vsphere/changelog.yml:6]: Use version: "1.14.0-next" as per this [issue](https://github.com/elastic/obs-integration-team/issues/22).

File: packages/vsphere/data_stream/resourcepool/manifest.yml (hunk 1 above)
Comment (niraj-elastic) [reply packages/vsphere/data_stream/resourcepool/manifest.yml:29]: Added screen shot in description of PR.

File: packages/vsphere/data_stream/resourcepool/_dev/test/pipeline/test-common-config.yml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+dynamic_fields:
+  "event.ingested": ".*"
```
Comment (ishleenk17) [inline packages/vsphere/data_stream/resourcepool/_dev/test/pipeline/test-common-config.yml:2]: Do we need this ?

File: packages/vsphere/data_stream/resourcepool/fields/fields.yml (hunk 3 above)
Comment (ishleenk17) [reply packages/vsphere/data_stream/resourcepool/fields/fields.yml]: unit missing in couple of byte fields

File: packages/vsphere/data_stream/resourcepool/_dev/test/pipeline/test-common-config.yml (hunk 5 above)
Comment (niraj-elastic) [reply packages/vsphere/data_stream/resourcepool/_dev/test/pipeline/test-common-config.yml:2]: no we can remove this file

File: packages/vsphere/changelog.yml [hunk 6]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.15.0-next"
+  changes:
+    - description: vSphere integration package with "resourcepool" data stream.
```
Comment (ishleenk17) [inline packages/vsphere/changelog.yml:4]: Updaete description similar to cluster DS.

File: packages/vsphere/_dev/build/docs/README.md [hunk 7]
```diff
@@ -45,6 +45,17 @@ Please refer to the following [document](https://www.elastic.co/guide/en/ecs/cur
 
 {{fields "datastore"}}
 
+### Resourcepool Metrics
+Resource pools in vSphere allow for the allocation and management of CPU and memory resources across groups of virtual machines.
+
+{{event "resourcepool"}}
```
Comment (ishleenk17) [inline packages/vsphere/_dev/build/docs/README.md:51]: Mention in datastreams

**Review by kush-elastic — APPROVED** [review]
LGTM!

**Review by lucian-ioan — APPROVED** [review]
LGTM

File: packages/vsphere/_dev/build/docs/README.md [hunk 8]
```diff
@@ -16,14 +16,16 @@ The vSphere integration collects logs and metrics.
...
 
 Data streams:
 - `log`: This data stream collects logs generated by VMware vSphere using a syslog daemon.
+- `cluster`: This data stream collects metrics from VMware vSphere, such as lists of datastores, hosts and networks associated with cluster.
 - `datastore`: This data stream collects datastore metrics from VMware vSphere, including performance statistics such as capacity, usage, read/write operations, latency, and throughput. These metrics are available in this data stream.
 - `host`: This data stream collects host metrics from VMware vSphere, including performance statistics such as CPU usage, memory usage, disk I/O, and network activity. These metrics are available in this data stream.
+- `resourcepool`: This data stream collects metrics from VMware vSphere, such as performance statistics such as CPU and memory usage, CPU and memory reservation, and CPU and memory limit. These metrics are available in this data stream.
```
Comment (ishleenk17) [inline packages/vsphere/_dev/build/docs/README.md:26]: This doesn'tcontain any performance metrics, right ?
Need to reword this

File: packages/vsphere/_dev/build/docs/README.md [hunk 9]
```diff
@@ -94,16 +107,16 @@ Please refer to the following [document](https://www.elastic.co/guide/en/ecs/cur
 
 {{fields "host"}}
 
-### Cluster Metrics
-Clusters in vSphere represent a group of ESXi hosts working together to optimize resource allocation, ensure high availability, and manage workloads efficiently.
+### Resourcepool Metrics
+Resource pools in vSphere allow for the allocation and management of CPU and memory resources across groups of virtual machines.
```
Comment (ishleenk17) [inline packages/vsphere/_dev/build/docs/README.md:111]: Is it valid only for virtual machines ?
Other entities are also part of it, right ?

Comment (niraj-elastic) [reply packages/vsphere/_dev/build/docs/README.md:111]: No other entities are collected in Resourcepool other than VMs.

File: packages/vsphere/_dev/build/docs/README.md [hunk 10]
```diff
@@ -23,7 +23,7 @@ Data streams:
 - `cluster`: This data stream collects metrics from VMware vSphere, such as lists of datastores, hosts and networks associated with cluster.
 - `datastore`: This data stream collects datastore metrics from VMware vSphere, including performance statistics such as capacity, usage, read/write operations, latency, and throughput. These metrics are available in this data stream.
 - `host`: This data stream collects host metrics from VMware vSphere, including performance statistics such as CPU usage, memory usage, disk I/O, and network activity. These metrics are available in this data stream.
-- `resourcepool`: This data stream collects metrics from VMware vSphere, such as performance statistics such as CPU and memory usage, CPU and memory reservation, and CPU and memory limit. These metrics are available in this data stream.
+- `resourcepool`: This data stream collects metrics from VMware vSphere, such as CPU and memory usage, CPU and memory reservation, and CPU and memory limit. These metrics are available in this data stream.
```
Comment (ishleenk17) [inline packages/vsphere/_dev/build/docs/README.md:26]:
```suggestion
- `resourcepool`: This data stream collects metrics from VMware vSphere, such as CPU and memory usage, CPU and memory reservation, and CPU and memory limit. 
```

Comment (ishleenk17) [reply packages/vsphere/_dev/build/docs/README.md:26]: Can we also make this deletion for other datastreams mentioned. Its redundant

**Review by ishleenk17 — APPROVED** [review]
Looks good!
Let the CI pass

**Pull request merged**